src/main/java/com/onlinebookstore/onlinebookstore/controller/BookController.java

```diff
@@ -28,21 +29,21 @@
 @RestController
 @RequestMapping(value = "api/books")
 public class BookController {
-
     private final BookService bookService;
 
     @GetMapping
     @Operation(summary = "Find All Books", description = "Find All Books in DB")
-    @PreAuthorize("hasRole('USER')")
+    @PreAuthorize("hasAnyRole('USER','ADMIN')")
     public List<BookDto> getAll(Authentication authentication, Pageable pageable) {
         User user = (User) authentication.getPrincipal();
         return bookService.findAll(user.getEmail(), pageable);
     }
 
     @GetMapping("/{id}")
     @Operation(summary = "Find Book by id", description = "Find Book by id")
-    @PreAuthorize("hasRole('USER')")
-    public BookDto findById(@PathVariable Long id) {
+    @PreAuthorize("hasAnyRole('USER','ADMIN')")
+    public BookDto findById(@PathVariable Long id,Authentication authentication) {

```

Comment:
Do you use authentication here?

Reply:
I speck to others and it turns out Admin also must be allowed to use get all and by id AS users but still they must be authorized